ReactWindows/ReactNative/Views/Image/ReactImageManager.cs

```diff
@@ -244,5 +276,63 @@ private void OnImageOpened(Border view)
                     view.GetTag(),
                     ReactImageLoadEvent.OnLoadEnd));
         }
+
+        /// <summary>
+        /// Set the source URI of the image.
+        /// </summary>
+        /// <param name="view">The image view instance.</param>
+        /// <param name="source">The source URI.</param>
+        private void SetUriFromSingleSource(Border view, string source)
+        {
+            var imageBrush = (ImageBrush)view.Background;
+            var tag = view.GetTag();
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactImageLoadEvent(
+                        tag,
+                        ReactImageLoadEvent.OnLoadStart));
+
+            var reference = _imageCache.Get(source);
+            var subscription = reference.LoadedObservable.Subscribe(
+                _ =>
+                {
+                    imageBrush.ImageSource = reference.Image;
+                    OnImageOpened(view);
+                },
+                _ => OnImageFailed(view));
+
+            var disposable = default(SerialDisposable);
+            if (!_disposables.TryGetValue(tag, out disposable))
+            {
+                disposable = new SerialDisposable();
+                _disposables.Add(tag, disposable);
+            }
+
+            disposable.Disposable = new CompositeDisposable
+            {
+                reference,
+                subscription,
+            };
+        }
+
+        /// <summary>
+        /// Chooses the uri with the size closest to the target image size. Must be called only after the
+        /// layout pass when the sizes of the target image have been computed, and when there are at least
+        /// two sources to choose from.
+        /// </summary>
+        /// <param name="view">The image view instance.</param>
+        private void SetUriFromMultipleSources(Border view)
+        {
+            var sources = default(List<KeyValuePair<string, double>>);
+            if (_imageSources.TryGetValue(view.GetTag(), out sources))
+            {
+                var targetImageSize = view.Width * view.Height;
+                var bestResult = sources.LocalMin((s) => Math.Abs(s.Value - targetImageSize));
```

Comment:
> Math.Abs(s.Value - targetImageSize) [](start = 56, length = 36)

Just checking, this selector function optimizes for the same result as (1 - s.Value / targetImageSize)?